master/resource/capacity_impl.go

```diff
@@ -0,0 +1,95 @@
+package resource
+
+import (
+	"sync"
+
+	"github.com/hanfei1991/microcosm/model"
+	"github.com/hanfei1991/microcosm/pb"
+	"github.com/hanfei1991/microcosm/pkg/errors"
+)
+
+// CapRescMgr implements ResourceMgr interface, and it uses node capacitiy as
+// alloction algorithm
+type CapRescMgr struct {
+	mu        sync.Mutex
+	executors map[model.ExecutorID]*ExecutorResource
+}
+
+func NewCapRescMgr() *CapRescMgr {
+	return &CapRescMgr{
+		executors: make(map[model.ExecutorID]*ExecutorResource),
+	}
+}
+
+// Register implements RescMgr.Register
+func (m *CapRescMgr) Register(id model.ExecutorID, capacity RescUnit) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	m.executors[id] = &ExecutorResource{
+		ID:       id,
+		Capacity: capacity,
+	}
+}
+
+// Allocate implements RescMgr.Allocate
+func (m *CapRescMgr) Allocate(tasks []*pb.ScheduleTask) (bool, *pb.TaskSchedulerResponse) {
+	return m.allocateTasksWithNaiveStrategy(tasks)
+}
+
+// Update implements RescMgr.Update
+func (m *CapRescMgr) Update(id model.ExecutorID, use RescUnit, status model.ExecutorStatus) error {
```

Comment:
When an executor is put offline, we should also remove it from resource manager.

Reply:
Good catch, addressed